lib/plugins/staticFiles.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+var assert = require('assert-plus');
+var errors = require('restify-errors');
+var path = require('path');
+var send = require('send');
+
+///--- Globals
+var MethodNotAllowedError = errors.MethodNotAllowedError;
+var NotAuthorizedError = errors.NotAuthorizedError;
+var ResourceNotFoundError = errors.ResourceNotFoundError;
+
+/**
+ * Serves static files, with API similar to expressjs
+ *
+ * @public
+ * @function serveStaticFiles
+ * @param    {String} directory - the directory to serve files from
+ * @param    {Object} opts - an options object, which is optional
```

Comment:
Is it worth calling this out as an options object meant explicitly for `send`? I think we do something similar with other plugins like mime.

Reply:
The options object is not explicitly for `send`, I added another functionality (similar to express) that allows us to set custom headers by passing a function under `options.setHeaders`. eg.
```
setHeaders: function setCustomHeaders(response, requestedPath, stat) {
              response.setHeader('restify-plugin-x', 'awesome');
          }
 })
```

Also, we explicitly override `options.root` before passing to `send`. So the other options might be similar to what `send` needs but they are not explicitly for `send`.

Reply:
Cool, makes sense. This might be paranoia but if it's not too expensive, you could think about making a new copy of opts when we call `send` but with only send specific options. It's nice to be able to separate internal opts from send opts. At the minimum - worth calling out in the jsdocs what's used by us vs used by `send`.